### ISSUE TITLE
Enable noImplicitAny for matrix-js-sdk and matrix-react-sdk

### ISSUE THREAD

**Issue by novocaine** [vector-im/element-web#21968]
### Your use case

https://www.typescriptlang.org/tsconfig#noImplicitAny

This will mitigate errors at runtime caused by incorrect types being passed.

**Comment by t3chguy** [comment]
js-sdk now has `strict` mode enabled, which includes noImplicitAny

**Comment by t3chguy** [comment]
`Found 1421 errors in 371 files.` to go

**Comment by Johennes** [comment]
@t3chguy is the plan for you to fix all of these yourself as part of this issue?

**Comment by t3chguy** [comment]
@Johennes yes

**Comment by Johennes** [comment]
Ouch, that seems like a lot then? 😬

I wonder if there's a reasonable way to split this up across multiple people? So that everyone puts in _some_ time rather than just one person having to fight through it alone.

**Comment by t3chguy** [comment]
Not without stepping on each other's feet and causing merge conflicts, given "fixing" a file often needs fixing the files it depends on too.

**Comment by Johennes** [comment]
Yeah, it does seem better at this point to put in time to fix this with a conscious effort rather than relying on PRs. I'm just a little worried about you having to fix a thousand issues all by yourself. Sounds like that might get quite annoying quickly. 😟

The issue of fixes in one file requiring further fixes in dependent files is tricky though. Could we have people work on their chunks sequentially? With 5 devs, each one could take ~50 files but we'd make sure to only have one person work on their subtask at a time?

**Comment by t3chguy** [comment]
@Johennes I've easily fixed 5x that already in previous efforts.

**Comment by Johennes** [comment]
> @Johennes I've easily fixed 5x that already in previous efforts.

Ugh. 😳 

Ok, well if you're happy to power through them, let's do that. 👍

Just keep in mind that we have 5 people in the immediate team now. So if we wanted to, we could easily organize a team effort here without having to depend on good will.

**Comment by t3chguy** [comment]
@Johennes if you have ideas on how to task those people with further tsc strict tasks in my absence next week to keep the momentum going then that'd be great to roll

**Comment by t3chguy** [comment]
Blocked on https://github.com/vector-im/element-web/issues/21967
